### PR TITLE
Feature #187 - Email notifications 3

### LINKED ISSUE #177: Email Notifications - Part 1
Parent issue: #176 

- [x] Since there are different types of emails, create some django choices `EmailTypes` with the choices `report, review_required, review_to_process`.

- [x] Create a model for the automatic email. This could contain the fields: `type` (choice of `EmailTypes`, should be unique), `body`, `subject`. In this way, it will then be easy to customise the text of the different emails in the admin.

- [x] Refactoring: for example creating a `notify` method to send emails.  

- [x] Modify the method https://github.com/maykinmedia/archiefvernietigingscomponent/blob/master/src/archiefvernietigingscomponent/destruction/models.py#L98 to send an email once someone is assigned to a list.

- [x] Modify the task https://github.com/maykinmedia/archiefvernietigingscomponent/blob/master/src/archiefvernietigingscomponent/destruction/tasks.py#L166 to make use of the configured email text/subject.

### LINKED ISSUE #178: Email Notifications - Part 2
Parent Issue: https://github.com/maykinmedia/archiefvernietigingscomponent/issues/176

- [x] Add a model to store user email-preferences `EmailPreferences`. Should have a relation to a user, and a choice of preferences, for example:
	- Email as soon as an action occurs
	- Never email
- [x] Add frontend to change email preferences
- [x] Add documentation about the emails

### LINKED ISSUE #187: Email notifications - Part 3
- [x] Add fixture with default email texts: https://github.com/maykinmedia/archiefvernietigingscomponent/issues/63#issuecomment-800246349
- [x] The default texts should be able to include variables, such as:

[naam gebruiker] - name of user who is receiving the mail
[https://etc] - link to vernietigingslijst
[naam vernietigingslijst] - name of the VL as given by the record manager
[naam archivaris] - also name of user archivaris
[gemeentenaam] - name of municipality

- [x] Documentation

### PR DESCRIPTION
**Pull request by SilviaAmAm**
Fixes #177 
Fixes #178 
Fixes #187 

Third part of the email notification feature

=> Depends on https://github.com/maykinmedia/archiefvernietigingscomponent/pull/198

### CODE REVIEW

File: src/archiefvernietigingscomponent/conf/locale/nl/LC_MESSAGES/django.po [hunk 1]
```diff
@@ -609,15 +622,108 @@ msgstr ""
...
 msgid "%(duration)s days"
 msgstr "%(duration)s dagen geleden"
 
+#: archiefvernietigingscomponent/emails/constants.py:7
+msgid "Report available"
+msgstr ""
+
+#: archiefvernietigingscomponent/emails/constants.py:8
+msgid "Review required"
+msgstr ""
```
Comment (joeribekker) [inline src/archiefvernietigingscomponent/conf/locale/nl/LC_MESSAGES/django.po:641]:
```suggestion
msgstr "Review verplicht"
```

File: src/archiefvernietigingscomponent/conf/locale/nl/LC_MESSAGES/django.po [hunk 2]
```diff
@@ -609,15 +622,108 @@ msgstr ""
...
+
+#: archiefvernietigingscomponent/emails/constants.py:9
+#, fuzzy
+#| msgid "Changes requested"
+msgid "Changes required"
+msgstr "Wijzigingen verzocht"
```
Comment (joeribekker) [inline src/archiefvernietigingscomponent/conf/locale/nl/LC_MESSAGES/django.po:647]: Remove fuzzy comments or it wont work.

File: src/archiefvernietigingscomponent/conf/locale/nl/LC_MESSAGES/django.po [hunk 3]
```diff
@@ -609,15 +622,108 @@ msgstr ""
...
+
+#: archiefvernietigingscomponent/emails/constants.py:13
+msgid "An action is required"
+msgstr "Wijzigingen verzocht"
```
Comment (joeribekker) [inline src/archiefvernietigingscomponent/conf/locale/nl/LC_MESSAGES/django.po:651]:
```suggestion
msgstr "Er wordt een actie verzocht"
```

File: src/archiefvernietigingscomponent/emails/models.py [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+        """
+        :param recipient: type User
+        :param destruction_list: type DestructionList:
+        :param report: type DestructionReport (optional)
+        :rtype: None
+        """
+        email = EmailMessage(
+            subject=self.subject,
+            body=self.compose_body(recipient, destruction_list, report),
+            from_email=settings.DEFAULT_FROM_EMAIL,
```
Comment (joeribekker) [inline src/archiefvernietigingscomponent/emails/models.py:80]: Let's make `from_email` configuratble in an existing solo config so we can configure the email address in the admin. Default can be  `archiefvernietigingscomponent@example.com`.

File: src/archiefvernietigingscomponent/fixtures/automatic_emails_defaults.json [hunk 5]
```diff
@@ -0,0 +1,36 @@
+[
+  {
+    "model": "emails.emailconfig",
+    "pk": 1,
+    "fields": {
+      "municipality": "Delft"
```
Comment (joeribekker) [inline src/archiefvernietigingscomponent/fixtures/automatic_emails_defaults.json:6]:
```suggestion
      "municipality": "Example"
```

File: doc/installation/configuration.rst [hunk 6]
```diff
@@ -300,3 +300,38 @@ Setting the domain
...
+
+A custom email subject and email body can be configured for each type of email. If no automatic emails are present
+in the admin in **Configuratie > Automatische emails**, then no emails will be sent.
+
+To load default automatic emails, a fixture is present. This can be loaded with the following command (from the
+``archiefvernietigingscomponent/`` directory):
+
+.. code-block:: bash
+
+    python src/manage.py loaddata src/archiefvernietigingscomponent/fixtures/automatic_emails_defaults.json
```
[the remark this answers is not in the document]
Comment (joeribekker) [reply doc/installation/configuration.rst:325]: And add this block to quickstart/demo I think (see where i added loading default roles as well).

**Review by joeribekker — CHANGES_REQUESTED** [review]

File: src/archiefvernietigingscomponent/conf/locale/nl/LC_MESSAGES/django.po (hunk 1 above)
Comment (SilviaAmAm) [reply src/archiefvernietigingscomponent/conf/locale/nl/LC_MESSAGES/django.po:641]: I think that the English for this is more along the line of "Review requested", so I'm not sure about "verplicht" :thinking:

Comment (joeribekker) [reply src/archiefvernietigingscomponent/conf/locale/nl/LC_MESSAGES/django.po:641]: `Review vereist`

File: src/archiefvernietigingscomponent/conf/locale/nl/LC_MESSAGES/django.po (hunk 3 above)
Comment (joeribekker) [reply src/archiefvernietigingscomponent/conf/locale/nl/LC_MESSAGES/django.po]: Een actie is vereist

**Review by joeribekker — APPROVED** [review]

**Pull request merged**